packages/node/src/behavior/state/validation/assertions.ts

```diff
@@ -62,3 +62,59 @@ export function assertArray(value: Val, path: SchemaErrorPath): asserts value is
         throw new DatatypeError(path, "an array", value);
     }
 }
+
+export const assertInt = {
+    /**
+     * Assertions for each integer type that is not nullable.
+     */
+    notNullable: {} as Record<string, typeof assertNumeric | undefined>,
+
+    /**
+     * Assertions for nullable integer types.
+     *
+     * These are separate from the "not nullable" assertions because Matter reserves a high or low value (for unsigned
+     * and signed, respectively) to indicate the field is null.
+     */
+    nullable: {} as Record<string, typeof assertNumeric | undefined>,
+};
+
+for (let i = 1n; i < 9n; i++) {
+    const intName = `int${i * 8n}`;
+    const uintName = `u${intName}`;
+
+    const unsignedUpperBound = 2n ** (i * 8n);
+    assertInt.notNullable[uintName] = createIntAssertion(uintName, 0n, unsignedUpperBound);
+    assertInt.nullable[uintName] = createIntAssertion(`nullable ${uintName}`, 0n, unsignedUpperBound - 1n);
+
+    const signedUpperBound = unsignedUpperBound / 2n;
+    const signedLowerBound = -signedUpperBound + 1n;
+    assertInt.notNullable[intName] = createIntAssertion(intName, signedLowerBound, signedUpperBound);
+    assertInt.nullable[intName] = createIntAssertion(`nullable ${intName}`, signedLowerBound + 1n, signedUpperBound);
+}
+
+function createIntAssertion(name: string, lowerBoundInclusive: bigint, upperBoundExclusive: bigint) {
```

Comment:
Why you do  it inclusive vs exclusive and then need to calculate again just for the error message, why not staying inclusive on both bounds?